test/test_alarm.cpp

```diff
@@ -294,60 +294,149 @@ TEST_F (AlarmUnitTest, AlarmCheckAddDiagItemToSummary)
       PNET_DIAG_CH_PROP_MAINT_FAULT);
 }
 
-
-TEST_F (AlarmUnitTest, AlarmCheckQueueHandling)
+TEST_F (AlarmUnitTest, AlarmCheckSendQueueHandling)
 {
-   pf_alarm_queue_t queue;
+   pf_alarm_send_queue_t queue;
    pf_alarm_data_t alarm_input;
    pf_alarm_data_t alarm_output;
    int ix = 0;
    int err = 0;
    const uint16_t SEQUENCE_START_NUMBER = 100;
 
    /* Prepare default data */
-   memset (&queue, 0, sizeof (queue));
    memset (&alarm_input, 0, sizeof (alarm_input));
    memset (&alarm_output, 0, sizeof (alarm_output));
+
+   err = pf_alarm_reset_send_queue (&queue);
+   EXPECT_EQ (err, 0);
    EXPECT_EQ (queue.count, 0);
 
    /* Fill the queue */
    for (ix = 0; ix < PNET_MAX_ALARMS; ix++)
    {
       alarm_input.sequence_number = SEQUENCE_START_NUMBER + ix;
-      err = pf_alarm_add_send_queue(&queue, &alarm_input);
+      err = pf_alarm_add_send_queue (&queue, &alarm_input);
       EXPECT_EQ (err, 0);
    }
    EXPECT_EQ (queue.count, PNET_MAX_ALARMS);
 
    /* Add to full queue */
    alarm_input.sequence_number += 1;
-   err = pf_alarm_add_send_queue(&queue, &alarm_input);
+   err = pf_alarm_add_send_queue (&queue, &alarm_input);
    EXPECT_EQ (err, -1);
    EXPECT_EQ (queue.count, PNET_MAX_ALARMS);
 
    /* Fetch from the queue */
    for (ix = 0; ix < PNET_MAX_ALARMS; ix++)
    {
-      err = pf_alarm_fetch_send_queue(&queue, &alarm_output);
+      err = pf_alarm_fetch_send_queue (&queue, &alarm_output);
+      EXPECT_EQ (err, 0);
+      EXPECT_EQ (alarm_output.sequence_number, SEQUENCE_START_NUMBER + ix);
+   }
+   EXPECT_EQ (queue.count, 0);
+
+   /* Fetch from empty queue */
+   err = pf_alarm_fetch_send_queue (&queue, &alarm_output);
+   EXPECT_EQ (err, -1);
+   EXPECT_EQ (queue.count, 0);
+
+   /* Reset the queue */
+   err = pf_alarm_add_send_queue (&queue, &alarm_input);
+   EXPECT_EQ (err, 0);
+   EXPECT_EQ (queue.count, 1);
+
+   err = pf_alarm_reset_send_queue (&queue);
+   EXPECT_EQ (err, 0);
+   EXPECT_EQ (queue.count, 0);
+
+   /* Wrap read_index and write_index, by adding and fetching a lot */
+   for (ix = 0; ix < PNET_MAX_ALARMS * 5; ix++)
+   {
+      alarm_input.sequence_number = SEQUENCE_START_NUMBER + ix;
+      err = pf_alarm_add_send_queue (&queue, &alarm_input);
+      EXPECT_EQ (err, 0);
+      EXPECT_EQ (queue.count, 1);
+
+      err = pf_alarm_fetch_send_queue (&queue, &alarm_output);
       EXPECT_EQ (err, 0);
+      EXPECT_EQ (queue.count, 0);
       EXPECT_EQ (alarm_output.sequence_number, SEQUENCE_START_NUMBER + ix);
    }
+}
+
+TEST_F (AlarmUnitTest, AlarmCheckReceiveQueueHandling)
+{
+   pf_alarm_receive_queue_t queue;
+   pf_apmr_msg_t input_frame;
+   pf_apmr_msg_t output_frame;
+   int ix = 0;
+   int err = 0;
+   const uint16_t POSITION_START_NUMBER = 100;
+   const uint64_t POINTER_START_NUMBER = 1000; // TODO uint64_t not portable
```

Comment:
Not sure what this is referring to but note that there is a uintptr_t type capable of holding a pointer on the target platform.

Reply:
Great, thanks!
Done.